src/configure/helper/telemetryHelper.ts

```diff
@@ -0,0 +1,66 @@
+import { IActionContext, ITelemetryReporter } from "vscode-azureextensionui";
+
+import { extensionVariables } from "../model/models";
+import { TracePoints } from "../resources/tracePoints";
+
+const uuid = require('uuid/v4');
+
+export class TelemetryHelper {
+    private actionContext: IActionContext;
+    private telemetryReporter: ITelemetryReporter;
+    private journeyId: string;
+    private command: string;
+    constructor(actionContext: IActionContext, command: string) {
+        this.actionContext = actionContext;
+        this.telemetryReporter = extensionVariables.reporter;
+        this.journeyId = uuid();
+        this.command = command;
+        this.setTelemetry(TracePoints.Command, command);
+        this.setTelemetry(TracePoints.JourneyId, this.journeyId);
+    }
+
+    public setTelemetry(key: string, value: string) {
+        if (key) {
+            this.actionContext.telemetry.properties[key] = value;
+        }
+    }
+
+    public setError(error: Error) {
+        this.actionContext.telemetry.properties.error = error.stack;
+        this.actionContext.telemetry.properties.errorMessage = error.message;
+    }
+
+    public setResult(result: 'Succeeded' | 'Failed' | 'Canceled', error?: Error, currentStep?: string) {
+        this.actionContext.telemetry.properties.result = result;
+        if (result === "Failed" && error) {
+            this.setError(error);
+        }
+        else if(result === 'Canceled' && currentStep) {
```

Comment:
is this necessary? We are already logging currentStep at major points. This can introduce unexpected step names as I am not sure what name will the called call with

Reply:
I kept this as an optional parameter. But yeah i can remove this.